optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -57,3 +57,86 @@ and supports <<explainingTheScore,justifications>>:
                         Talk::getDurationInMinutes);
     }
 ----
+
+[[constraintStreamsGroupingAndCollectors]]
+== Grouping and Collectors
+
+In the constraint streams framework, grouping is a mechanism to group items in a stream according to user-provider
+criteria, similar to what a `GROUP BY` SQL clause does. Additionally, some grouping operations also accept one or more
+`Collector`s, which provide various aggregation functions.
+
+For example, the following code snippet will first group all processes by the computer they run on, sum up all the
+power required by the processes on that computer using the `ConstraintCollectors.sum(...)` collector, and finally
+penalize every computer whose processes consume more power than is available.
+
+[source,java,options="nowrap"]
+----
+    private Constraint requiredCpuPowerTotal(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(CloudProcess.class)
+                .groupBy(CloudProcess::getComputer, sum(CloudProcess::getRequiredCpuPower))
+                .penalize("requiredCpuPowerTotal",
+                        HardSoftScore.ONE_HARD,
+                        (computer, requiredCpuPower) -> requiredCpuPower - computer.getCpuPower());
+    }
+----
+
+OptaPlanner provides several collector implementations you can use in your grouping constraint streams. This chapter
```

Comment:
"OptaPlanner provides" -> there are
"you can use" -> "to"
"in your grouping constraint streams" -> singular is more accurate. Per one, you can still use multiple collectors.
->
"There are several collectors out-of-the-box when grouping a constraint stream."